fields.yml

```diff
@@ -381,35 +381,55 @@
             Unique ID to describe the event.
           example: 8a4f500d
     
+        - name: kind
+          level: core
+          type: keyword
+          description: >
+            The kind of the event.
+    
+            This gives information about what type of information the event
+            contains, without being specific to the contents of the event.  Examples
+            are `event`, `state`, `alarm`.
```

Comment:
Instead of `event`, perhaps we should use `log` here? For me all 3 kind are an `event` (based on ECS).

Reply:
@cwurm Also mentioned `action` here as an option.

Reply:
`log` is not great, btw, in Auditbeat we have events like "process start" that we get from a netlink socket, not from a "log".

Reply:
Yes, I agree "event" is better than "log" for this.

Reply:
I think the definition should be a bit more forceful in that this field **must** contain one of these values. The goal here is super broad categorization of how this document/event came about. So the values have to be predictable:

- event: just reporting something that was observed in real-time (e.g. syslog message, log entry, event on a bus such as ebpf).
- state: generating an update on the state of something on a regular schedule (list of running processes, installed packages, current CPU usage).
  * So for example, metrics from metricbeat should all use `event.kind:state`.
- alarm: alerting on the fact that something unwanted happened, or a state changed enough to warrant attention from someone. Could be triggered by a static threshold, ML, a rules engine, etc.

What we want to avoid here is to have people map arbitrary values from their event sources to this field, and we end up with lots of unpredictable values in `event.kind`.

Reply:
Of course the corollary is that if people determine that their stuff doesn't fall into one of these 3, we want them to open a GH issue to discuss it, and potentially grow that list.

Reply:
I added a warning like for the others. I'm not sure we want to define `event`, `state`, `alarm`, etc just yet, there will be more than that and we'd do a poor job if we try to do it by the freeze date.

Reply:
I think the defined list of values later on should go into a different document and potentially we have 1 doc per use case and the values for each use case might be different. 

`log` for me does not mean it must come from a file. I'm more thinking of the historical definition `an official record of events during the voyage of a ship or aircraft`.

Reply:
> @cwurm Also mentioned `action` here as an option.

I think it was `state_restart`. I wanted a way to distinguish between a state that is triggered periodically (e.g. every 1h) and one triggered by a restart of (in this case) Auditbeat. One could imagine other reasons it was sent (e.g. a threshold was breached, or that too many changes accumulated). If we had something like an `event.trigger` field that captures _why_ a document was sent that would also work - but maybe that's one field too many as well.